src/github/mod.rs

```diff
@@ -0,0 +1,220 @@
+        owner,
+        name,
+        branch,
+    } = repository;
+
+    let url = format!("https://github.com/{owner}/{name}/archive/{branch}.zip");
+    let response = reqwest::get(url).await?.bytes().await?;
+
+    let reader = std::io::Cursor::new(response);
+    let mut archive = zip::ZipArchive::new(reader)?;
+
+    let files: Vec<File> = (0..archive.len())
+        .filter_map(|file| {
+            let mut file = archive.by_index(file).unwrap();
+            let file_path = file.name().split_once('/').unwrap().1.to_string();
+
+            if file.is_file() && should_index(&file_path) {
+                let mut content = String::new();
+                let length = content.len();
+                //Fails for non UTF-8 files
+                match file.read_to_string(&mut content) {
+                    Ok(_) => Some(File {
+                        path: file_path,
+                        content,
+                        length,
+                    }),
+                    Err(_) => None,
+                }
+            } else {
+                None
+            }
+        })
+        .collect();
+    Ok(files)
+}
+
+pub async fn fetch_file_content(repository: &Repository, path: &str) -> Result<String> {
+    let Repository {
+        owner: repo_owner,
+        name: repo_name,
+        branch: repo_branch,
+    } = repository;
+    let url =
+        format!("https://raw.githubusercontent.com/{repo_owner}/{repo_name}/{repo_branch}/{path}");
+    let response = reqwest::get(url).await?;
+    if response.status() == reqwest::StatusCode::OK {
+        let content = response.text().await?;
+        Ok(content)
+    } else {
+        Err(anyhow::anyhow!("Unable to fetch file content"))
+    }
+}
+
+const IGNORED_EXTENSIONS: &[&str] = &[
+    "bpg", "eps", "pcx", "ppm", "tga", "tiff", "wmf", "xpm", "svg", "ttf", "woff2", "fnt", "fon",
+    "otf", "pdf", "ps", "dot", "docx", "dotx", "xls", "xlsx", "xlt", "lock", "odt", "ott", "ods",
+    "ots", "dvi", "pcl", "mod", "jar", "pyc", "war", "ear", "bz2", "xz", "rpm", "coff", "obj",
+    "dll", "class", "log",
+];
+
+const IGNORED_DIRECTORIES: &[&str] = &[
+    "vendor",
+    "dist",
+    "build",
+    "target",
+    "bin",
+    "obj",
+    "node_modules",
+    "debug",
+];
+
+pub fn should_index(path: &str) -> bool {
+    !(IGNORED_EXTENSIONS.iter().any(|ext| path.ends_with(ext))
+        || IGNORED_DIRECTORIES.iter().any(|dir| path.contains(dir)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_fetch_repo_files() {
+        let repository = Repository {
+            owner: "open-sauced".to_string(),
+            name: "ai".to_string(),
+            branch: "beta".to_string(),
+        };
+
+        let result = fetch_repo_files(&repository).await;
+
+        // Assert that the function returns a Result containing a vector of File
+        assert!(result.is_ok());
+        let files = result.unwrap();
+        assert!(files.len() > 0);
```

Comment:
Nit: `cargo clippy` wants to make this change:

```diff
-        assert!(files.len() > 0);
+        assert!(!files.is_empty());
```

Reply:
https://github.com/open-sauced/repo-query/pull/1/commits/3ec893a487f6c806e750ec5e8e398d81024d9fd6